src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -3513,6 +3516,11 @@ TablePartitioningSupportsDistinct(List *tableNodeList, MultiExtendedOp *opNode,
 			Var *tablePartitionColumn = tableNode->partitionColumn;
 			bool groupedByPartitionColumn = false;
 
+			if (aggregateType == AGGREGATE_COUNT)
+			{
+				tableDistinctSupported = true;
```

Comment:
At this point, is there any reason not to just pull expressions like sum(distinct) to master?

Reply:
Not really, we should be able to pull any distinct expression to coordinator.

There was an issue with pulling up a non-column distinct like count(distinct 1) type expression where I spent most of my time working on this PR. We should also address that with more generalized aggregate distinct work.